lib/myxql/protocol/server_error_codes.ex

```diff
@@ -1,4483 +1,36 @@
-    {13191, :ER_IB_MSG_1275},
-    {13192, :ER_TARGET_TS_UNENCRYPTED},
-    {13193, :ER_IB_MSG_1276},
-    {13194, :ER_IB_MSG_1277},
-    {13195, :ER_IB_MSG_1278},
-    {13196, :ER_IB_MSG_1279},
-    {13197, :ER_IB_MSG_1280},
-    {13198, :ER_IB_MSG_1281},
-    {13199, :ER_IB_MSG_1282},
-    {13200, :ER_IB_MSG_1283},
-    {13201, :ER_IB_MSG_1284},
-    {13202, :ER_CANT_SET_ERROR_SUPPRESSION_LIST_FROM_COMMAND_LINE},
-    {13203, :ER_INVALID_VALUE_OF_BIND_ADDRESSES},
-    {13204, :ER_RELAY_LOG_SPACE_LIMIT_DISABLED},
-    {13205, :ER_GRP_RPL_ERROR_GTID_SET_EXTRACTION},
-    {13206, :ER_GRP_RPL_MISSING_GRP_RPL_ACTION_COORDINATOR},
-    {13207, :ER_GRP_RPL_JOIN_WHEN_GROUP_ACTION_RUNNING},
-    {13208, :ER_GRP_RPL_JOINER_EXIT_WHEN_GROUP_ACTION_RUNNING},
-    {13209, :ER_GRP_RPL_CHANNEL_THREAD_WHEN_GROUP_ACTION_RUNNING},
-    {13210, :ER_GRP_RPL_APPOINTED_PRIMARY_NOT_PRESENT},
-    {13211, :ER_GRP_RPL_ERROR_ON_MESSAGE_SENDING},
-    {13212, :ER_GRP_RPL_CONFIGURATION_ACTION_ERROR},
-    {13213, :ER_GRP_RPL_CONFIGURATION_ACTION_LOCAL_TERMINATION},
-    {13214, :ER_GRP_RPL_CONFIGURATION_ACTION_START},
-    {13215, :ER_GRP_RPL_CONFIGURATION_ACTION_END},
-    {13216, :ER_GRP_RPL_CONFIGURATION_ACTION_KILLED_ERROR},
-    {13217, :ER_GRP_RPL_PRIMARY_ELECTION_PROCESS_ERROR},
-    {13218, :ER_GRP_RPL_PRIMARY_ELECTION_STOP_ERROR},
-    {13219, :ER_GRP_RPL_NO_STAGE_SERVICE},
-    {13220, :ER_GRP_RPL_UDF_REGISTER_ERROR},
-    {13221, :ER_GRP_RPL_UDF_UNREGISTER_ERROR},
-    {13222, :ER_GRP_RPL_UDF_REGISTER_SERVICE_ERROR},
-    {13223, :ER_GRP_RPL_UDF_ERROR},
-    {13224, :ER_CURRENT_PASSWORD_NOT_REQUIRED},
-    {13225, :ER_INCORRECT_CURRENT_PASSWORD},
-    {13226, :ER_MISSING_CURRENT_PASSWORD},
-    {13227, :ER_SERVER_WRONG_VALUE_FOR_VAR},
-    {13228, :ER_COULD_NOT_CREATE_WINDOWS_REGISTRY_KEY},
-    {13229, :ER_SERVER_GTID_UNSAFE_CREATE_DROP_TEMP_TABLE_IN_TRX_IN_SBR},
-    {13230, :ER_SECONDARY_ENGINE},
-    {13231, :ER_SECONDARY_ENGINE_DDL},
-    {13232, :ER_NO_SESSION_TEMP},
-    {13233, :ER_XPLUGIN_FAILED_TO_SWITCH_SECURITY_CTX},
-    {13234, :ER_RPL_GTID_UNSAFE_ALTER_ADD_COL_WITH_DEFAULT_EXPRESSION},
-    {13235, :ER_UPGRADE_PARSE_ERROR},
-    {13236, :ER_DATA_DIRECTORY_UNUSABLE},
-    {13237, :ER_LDAP_AUTH_USER_GROUP_SEARCH_ROOT_BIND},
-    {13238, :ER_PLUGIN_INSTALL_ERROR},
-    {13239, :ER_PLUGIN_UNINSTALL_ERROR},
-    {13240, :ER_SHARED_TABLESPACE_USED_BY_PARTITIONED_TABLE},
-    {13241, :ER_UNKNOWN_TABLESPACE_TYPE},
-    {13242, :ER_WARN_DEPRECATED_UTF8_ALIAS_OPTION},
-    {13243, :ER_WARN_DEPRECATED_UTF8MB3_CHARSET_OPTION},
-    {13244, :ER_WARN_DEPRECATED_UTF8MB3_COLLATION_OPTION},
-    {13245, :ER_SSL_MEMORY_INSTRUMENTATION_INIT_FAILED}
+    {1792, :ER_CANT_EXECUTE_IN_READ_ONLY_TRANSACTION}
   ]
 
-  @spec code_to_name(integer()) :: atom()
-  def code_to_name(code)
+  # TODO: use Application.compile_env/3 when we require Elixir v1.10
+  codes = codes ++ Application.get_env(:myxql, :extra_error_codes, [])
 
   for {code, name} <- codes do
```

Comment:
```suggestion
  for {code, name} <- Enum.uniq(codes) do
```

In case someone adds a code that already exists (or we case we add a code in the future the user was using).